tasks/skylab/HISAT2.wdl

```diff
@@ -133,21 +153,36 @@ task HISAT2RSEM {
   }
 
   command {
+
     set -e
 
-    # fix names if necessary.
-    if [[ "${fastq1}" != *.fastq.gz ]]; then
-        FQ1=${fastq1}.fastq.gz
-        mv ${fastq1} ${fastq1}.fastq.gz
-    else
-        FQ1=${fastq1}
+   # fix names if necessary
+    if (file ~{fastq1} | grep -q compressed); then
+        if [[ ~{fastq1} != *.gz ]]; then
+            if [[ ~{fastq1} != *.fastq ]]; then
+                FQ1=~{fastq1}.fastq.gz
+                mv ~{fastq1} ~{fastq1}.fastq.gz
+            else
+                FQ1=~{fastq1}.gz
+                mv ~{fastq1} ~{fastq1}.gz
+            fi
+        else
+            FQ1="~{fastq1}"
+        fi
     fi
```

Comment:
Need the same check here

Reply:
I committed all of the changes together.